modules/tinymce/src/plugins/table/test/ts/browser/ui/TableBorderWidthTest.ts

```diff
@@ -0,0 +1,196 @@
+          '</tr>' +
+        '</tbody>' +
+      '</table>'
+    );
+
+    TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1);
+  };
+
+  const assertMultipleCellsStructureHasWidth = (editor: Editor) => {
+    TinyAssertions.assertContent(editor, '<table>' +
+      '<tbody>' +
+        '<tr>' +
+          '<td style="border-width: 1px;">Filler</td>' +
+          '<td style="border-width: 1px;">Filler</td>' +
+        '</tr>' +
+        '<tr>' +
+          '<td style="border-width: 1px;">Filler</td>' +
+          '<td style="border-width: 1px;">Filler</td>' +
+        '</tr>' +
+      '</tbody>' +
+    '</table>');
+  };
+
+  const assertMultipleCellsStructureIsWithoutWidth = (editor: Editor) => {
+    TinyAssertions.assertContent(editor, '<table>' +
+      '<tbody>' +
+        '<tr>' +
+          '<td>Filler</td>' +
+          '<td>Filler</td>' +
+        '</tr>' +
+        '<tr>' +
+          '<td>Filler</td>' +
+          '<td>Filler</td>' +
+        '</tr>' +
+      '</tbody>' +
+    '</table>');
+  };
+
+  const openMenu = (editor: Editor) =>
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Border width"]');
+
+  const closeMenu = (editor) =>
+    TinyUiActions.keydown(editor, Keys.escape());
+
+  const pAssertMenuPresence = async (editor: Editor, label: string, expected: Record<string, number>, container: SugarElement<HTMLElement>) => {
+    openMenu(editor);
+    await Waiter.pTryUntil('Ensure the correct values are present', () =>
+      Assertions.assertPresence(label, expected, container)
+    );
+    closeMenu(editor);
+  };
+
+  const pAssertNoCheckmarksInMenu = async (editor: Editor, container: SugarElement<HTMLElement>) => {
+    const expected = {
+      '.tox-menu': 1,
+      '.tox-collection__item[aria-checked="true"]': 0,
+      '.tox-collection__item[aria-checked="false"]': 2,
+      'div[title="None"]': 1,
+      'div[title="1PX"]': 1,
+    };
+
+    await pAssertMenuPresence(editor, 'Menu should open, but not have any checkmarks', expected, container);
+  };
+
+  const pSetBorderWidth = async (editor: Editor, toggleOn: boolean) => {
+    openMenu(editor);
+    await TinyUiActions.pWaitForUi(editor, `div[title="${toggleOn ? '1PX' : 'None'}"]`);
+    TinyUiActions.clickOnUi(editor, `div[title="${toggleOn ? '1PX' : 'None'}"]`);
+    closeMenu(editor);
+  };
+
+  const pAssertCheckmark = async (editor: Editor, sugarContainer: SugarElement<HTMLElement>) => {
+    const expected = {
+      '.tox-menu': 1,
+      '.tox-collection__item[aria-checked="true"][title="1PX"]': 1,
+      '.tox-collection__item[aria-checked="false"]': 1,
+      '.tox-collection__item[aria-checked="true"]': 1,
+    };
+    await pAssertMenuPresence(editor, 'There should be a checkmark', expected, sugarContainer);
+  };
+
+  it('TINY-7478: Ensure the table border width adds and removes it as expected with a single cell', async () => {
+    const editor = hook.editor();
+    const sugarContainer = SugarBody.body();
```

Comment:
I _think_ this can be declared outside the `it` blocks and used by the helper functions directly. So you don't need to pass it around

Reply:
I generally would advise against that, as while unlikely it's possible things could change between when the describe runs and the `it` block executes. It would be fine for this case, but if you were getting the editor body or something in an iframe it could have changed.

Reply:
Due to the restructuring this isn't really valid anymore.